### PR TITLE
fix: adding .git folder to the folder should propagate okteto git commit

### PR DESCRIPTION
**Pull request by jLopezbarb**
# Proposed changes

Fixes https://github.com/okteto/app/issues/6192

- If `OKTETO_GIT_COMMIT` is present in the remote deploy/destroy we should not try to get if the project is clean or if the project has the right hash
- Move git repo controller to its own file
- If the code is executed inside a remote deploy/destroy
  - Check the `OKTETO_GIT_COMMIT` to skip the `.git` project access 
  - If executed locally check it in the `.git` folder

### CODE REVIEW

File: cmd/deploy/deploy.go [hunk 1]
```diff
@@ -297,7 +297,7 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 		Name:       deployOptions.Name,
 		Namespace:  deployOptions.Manifest.Namespace,
 		Repository: os.Getenv(model.GithubRepositoryEnvVar),
```
Comment (teresaromero) [inline cmd/deploy/deploy.go:299]: moving this to constants would make sense?

Comment (jLopezbarb) [reply cmd/deploy/deploy.go:299]: It makes sense but I would avoid doing it in this PR to keep it as isolated as possible. I'd create an issue to move all the model constants to its own package later 😄

File: pkg/repository/repository.go [hunk 2]
```diff
@@ -30,111 +29,42 @@ type Repository struct {
...
-	worktree *git.Worktree
-}
-
-func (ogr oktetoGitWorktree) Status() (gitStatusInterface, error) {
-	status, err := ogr.worktree.Status()
-	if err != nil {
-		return nil, err
-	}
-	return oktetoGitStatus{status: status}, nil
+	repoCtrl repositoryInterface
```
Comment (teresaromero) [inline pkg/repository/repository.go:32]:
```suggestion
	control repositoryInterface
```

nit

File: pkg/repository/repository.go [hunk 3]
```diff
@@ -30,111 +29,42 @@ type Repository struct {
...
 }
 
+// NewRepository creates a repository controller
 func NewRepository(path string) Repository {
 	url, err := giturls.Parse(path)
 	if err != nil {
 		oktetoLog.Infof("could not parse url: %w", err)
 	}
+
+	var repoCtrl repositoryInterface = newGitRepoController()
```
Comment (teresaromero) [inline pkg/repository/repository.go:47]:
```suggestion
	var controller repositoryInterface = newGitRepoController()
```

**Review by teresaromero — APPROVED** [review]
minor coments, probably lacking context. Overall looks good.

File: pkg/repository/remote.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repository
+
+type oktetoInsideRemoteDeployRepositoryController struct {
```
[the remark this answers is not in the document]
Comment (AdrianPedriza) [reply pkg/repository/remote.go:16]: maybe `oktetoRemoteRepoController`?

**Pull request merged**